core/Objects/ProcessStartInfoCLI.cs

```diff
@@ -0,0 +1,108 @@
+using System.Collections.Specialized;
+using System.Diagnostics;
+
+namespace Cmf.CLI.Core.Objects
+{
+    /// <summary>
+    ///
+    /// </summary>
+    public class ProcessStartInfoCLI : IProcessStartInfoCLI
+    {
+        private ProcessStartInfo ps;
+        public Process Process { get; set; }
+
+        public ProcessStartInfoCLI()
+        {
+            this.ps = new ProcessStartInfo();
+        }
+
+        public ProcessStartInfoCLI(string fileName)
+        {
+            this.ps = new ProcessStartInfo(fileName);
+        }
+
+        public ProcessStartInfoCLI(string fileName, string arguments)
+        {
+            this.ps = new ProcessStartInfo(fileName, arguments);
+        }
+
+        public Process Start()
+        {
+            return StartProcess();
+        }
+
+        public void AddEvenErrorDataReceived(DataReceivedEventHandler handler)
+        {
+            this.Process.ErrorDataReceived += handler;
+        }
+
+        public void AddEventOutDataReceived(DataReceivedEventHandler handler)
+        {
+            this.Process.OutputDataReceived += handler;
+        }
+
+        protected Process StartProcess()
+        {
+            this.Process = System.Diagnostics.Process.Start(this.ps);
+            return this.Process;
+        }
+
+        public void BeginOutputReadLine()
+        {
+            this.Process.BeginOutputReadLine();
+        }
+
+        public void BeginErrorReadLine()
+        {
+            this.Process.BeginErrorReadLine();
+        }
+
+        public void WaitForExit()
+        {
+            this.Process.WaitForExit();
+        }
+
+        public void Dispose()
```

Comment:
this class should probably implement IDisposable?

Reply:
it implements indirectly, through the IProcessStartInfoCLI